README/RCodingStyle.hxx

```diff
@@ -0,0 +1,251 @@
+/// \file RCodingStyle.hxx
+/// \author Axel Naumann <axel@cern.ch>
+/// \author Other Author <other@author>
+/// \date 2018-07-24 or \date July, 2018
+// The above entries are mostly for giving some context.
+// The "author" field gives a hint whom to contact in case of questions, also
+// from within the team. The date shows whether this is ancient or only part
+// of the latest release. It's the date of creation / last massive rewrite.
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+// This file demonstrates the coding styles to be used for new ROOT files.
+// See the accompanying source file for context.
+
+// File names correspond to the name of the main class that they contain.
+// Our users will expect that a file called "RCodingStyle.hxx" contains a class
+// named `RCodingStyle`.
+
+// All files start with a Doxygen header, followed by the copyright statement.
+// The "-year" part is kept up to date by a script.
+
+// The file must contain no trailing whitespace including no indentation-only lines.
+// Except for extreme cases, line breaks should happen between 80 and 120 characters.
+
+// The first non-comment, non-empty line must be the #include guard.
+// Acceptable forms are:
+//   ROOT_RCodingStyle, ROOT_RCodingStyle_hxx, ROOT_RCodingStyle_Hxx,
+//   ROOT_RCODINGSTYLE, ROOT_RCODINGSTYLE_HXX
+// The first is preferred.
+#ifndef ROOT_RCodingStyle
+
+// Especially within header files, using forward declarations instead of
+// including the relevant headers wherever possible.
+// EXCEPTION: standard library classes must be #included, not forward declared.
+// All files must be stand-alone: all types must be available through a
+// forward declaration or an #include; it is not acceptable to rely on an
+// included header to itself include another header needed by the file.
+
+// Include ROOT's headers first, in alphabetical order:
+//   Rationale: make sure that ROOT headers are standalone.
+// Do not use #include guards at the point of `#include`.
+//   Rationale: compilers optimize this themselves, this is interfering with their optimization.
+// Use <>, not "".
+//   Rationale: "" looks at ./ first; as ROOT headers are included by user code
+//   that adds context fragility.
+#include <ROOT/TSeq.hxx>
+// Include v6 ROOT headers last.
+// You're welcome to state what's used from a header, i.e. why it's included.
```

Comment:
In practice, this might be outdated the second time a file is touched.

Reply:
Fully agree!